### ISSUE TITLE
TS error: Argument of type 'Stream<ChatCompletionChunk>' is not assignable to parameter of type 'Response | AsyncIterableOpenAIStreamReturnTypes

### ISSUE THREAD

**Issue by francescogior** [vercel/ai#610]
### Description

 "openai": "4.11.0"
 "ai": "2.2.13",

Passing openai `response` to `OpenAIStream` causes TS type error.

### Code example

```
import { OpenAIStream, StreamingTextResponse } from 'ai';
import OpenAI from 'openai';

const openai = new OpenAI({ apiKey: process.env.OPENAI_API_KEY });

const response = await openai.chat.completions.create({
...

const stream = OpenAIStream(response, {
...

```

causes

```
Type 'Stream<ChatCompletionChunk>' is not assignable to type 'AsyncIterable<Completion>'.
    The types returned by '[Symbol.asyncIterator]().next(...)' are incompatible between these types.
      Type 'Promise<IteratorResult<ChatCompletionChunk, any>>' is not assignable to type 'Promise<IteratorResult<Completion, any>>'.
        Type 'IteratorResult<ChatCompletionChunk, any>' is not assignable to type 'IteratorResult<Completion, any>'.
          Type 'IteratorYieldResult<ChatCompletionChunk>' is not assignable to type 'IteratorResult<Completion, any>'.
            Type 'IteratorYieldResult<ChatCompletionChunk>' is not assignable to type 'IteratorYieldResult<Completion>'.
              Type 'ChatCompletionChunk' is not assignable to type 'Completion'.
```

**Comment by francescogior** [comment]
> I had the exact same issue just a moment ago with my new project install ai openAI with yarn. It seems like a version problem as I just copied files from older project and the error was resolved.
> 
> I took a look and found this difference in yarn-lock file. Hope it can help.

Actually I was able to fix it by downgrading `openai` to 4.10.
Looks like there should be a strict peer dependency to avoid this? 🤔

**Comment by MaxLeiter** [comment]
This should be fixed in 2.2.14. Thank you all for the assistance!

**Comment by doodirock** [comment]
Still happening for me with:

1. "openai": "^4.14.2"
2. "ai": "^2.2.20"

Any ideas?

**Comment by zamaVR** [comment]
Still seeing this issue with `ai 2.2.14` and `openai 4.20.0`

```
Argument of type 'Stream<ChatCompletionChunk>' is not assignable to parameter of type 'Response | AsyncIterableOpenAIStreamReturnTypes'.
  Type 'Stream<ChatCompletionChunk>' is not assignable to type 'AsyncIterable<Completion>'.
    The types returned by '[Symbol.asyncIterator]().next(...)' are incompatible between these types.
      Type 'Promise<IteratorResult<ChatCompletionChunk, any>>' is not assignable to type 'Promise<IteratorResult<Completion, any>>'.
        Type 'IteratorResult<ChatCompletionChunk, any>' is not assignable to type 'IteratorResult<Completion, any>'.
          Type 'IteratorYieldResult<ChatCompletionChunk>' is not assignable to type 'IteratorResult<Completion, any>'.
            Type 'IteratorYieldResult<ChatCompletionChunk>' is not assignable to type 'IteratorYieldResult<Completion>'.
              Type 'ChatCompletionChunk' is not assignable to type 'Completion'.ts(2345)
```

**Comment by lgrammel** [comment]
## Identify issue type

Identified as **Bug**  
Confidence: **High**

### Rationale
The issue reports a concrete type incompatibility in an existing `OpenAIStream` integration, with a version-dependent workaround and an attempted SDK fix. This is broken existing behavior rather than a request for new capability.

### Evidence
- Passing OpenAI 4.11.0's `Stream<ChatCompletionChunk>` to `OpenAIStream` in AI SDK 2.2.13 produces a TypeScript incompatibility error.
- Downgrading OpenAI to 4.10 was reported as a workaround, indicating a version compatibility regression.
- A maintainer stated it should be fixed in AI SDK 2.2.14, but later users reported the same error with 2.2.14 and OpenAI 4.20.0.
